Android/src/org/droidplanner/android/activities/helpers/SuperUI.java

```diff
@@ -256,7 +249,20 @@ public void onNo() {
                 SlideToUnlockDialog unlockDialog = SlideToUnlockDialog.newInstance("disable vehicle", new Runnable() {
                     @Override
                     public void run() {
-                        dpApp.getDrone().arm(false);
+                        /* If the vehicle is a Copter running a firmware older than 3.3,
+                           set the vehicle to stabilize.  Otherwise, leave the mode alone.
+                           Then send the emergency disarm command.
+                         */
+                        Drone drone = dpApp.getDrone();
+                        Type droneType = drone.getAttribute(AttributeType.TYPE);
+                        if(droneType !=null){
+                            if(droneType.getDroneType() == (Type.TYPE_COPTER) &&
+                                    !droneType.getFirmwareVersion().startsWith("APM:Copter V3.3") &&
+                                    !droneType.getFirmwareVersion().startsWith("APM:Copter V3.4")){
+                                DroneStateApi.setVehicleMode(drone, VehicleMode.COPTER_STABILIZE);
+                            }
+                        }
+                        DroneStateApi.arm(drone, false, true);
```

Comment:
@squilter the version check logic can be removed. It's handled by 3DR Services in https://github.com/DroidPlanner/DroneKit-Android/blob/371cfd568046fc772726f9af3b2baf3e9e1dd866/ServiceApp/src/org/droidplanner/services/android/api/DroneApiUtils.java#L682-L748